test/lib/y2firewall/clients/auto_test.rb

```diff
@@ -33,11 +33,70 @@
     allow(subject).to receive(:importer).and_return(importer)
   end
 
+  describe "#general_summary" do
+    it "creates a text which includes firewalld overview" do
+      expect(firewalld).to receive(:running?).and_return(true)
+      expect(firewalld).to receive(:enabled?).and_return(true)
+      expect(firewalld).to receive(:default_zone).and_return("public")
+
+      summary = subject.send(:general_summary)
+
+      expect(summary).to match(/Running/)
+      expect(summary).to match(/Enabled/)
+      expect(summary).to match(/Default zone/)
+      expect(summary).to match(/Defined zones/)
+    end
+  end
+
+  describe "#zone_summary" do
+    RELATIONS = {
+      interfaces: ["eth0", "eth1"],
+      services:   ["ssh", "ftp"],
+      protocols:  ["udp", "tcp"],
+      ports:      ["80"]
+    }.freeze
+
+    def define_zone(name: nil, relation: nil, values: nil)
+      return nil if name.nil? || name.empty?
+      return nil if !Y2Firewall::Firewalld::Zone.instance_methods(false).include?(relation)
+
+      zone = Y2Firewall::Firewalld::Zone.new(name: name)
+      expect(zone).to receive(relation).and_return(values)
+
+      zone
+    end
+
+    it "empty zone returns empty description" do
```

Comment:
what about using a context
```ruby
  context "when the zone does not have any relation"
    it "returns an empty description" do
    end
  end
```

or something similiar

Reply:
in fact I was thinking of it. But as I had just one test to put in I didn't do it. However, can do it so no problem.